tests/integration/models/reddit/test_subreddit.py

```diff
@@ -9,6 +11,45 @@
 
 class TestSubreddit(IntegrationTest):
     @mock.patch('time.sleep', return_value=None)
+    def test_create(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette('TestSubreddit.test_create'):
+            # These values are hard-coded for the benefit of the cassettes.
+            new_name = 'PRAW_cvolgqjmge'
+            nonexistent_subreddit = 'PRAW_gbchyuppzp'
+
+            # Extra kwargs that are static throughout the test
+            kwargs = {'subreddit_type': 'public',
+                      'wikimode': 'disabled'}
+
+            with pytest.raises(Redirect) as excinfo:
+                # Redirects to /subreddits/search
+                next(self.reddit.subreddit(nonexistent_subreddit).new())
+
+            with pytest.raises(APIException) as excinfo:
+                self.reddit.subreddit.create('redditdev', title='x',
+                                             link_type='any', **kwargs)
+            assert excinfo.value.error_type == 'SUBREDDIT_EXISTS'
+
+            with pytest.raises(APIException) as excinfo:
+                # Not supplying required field title.
+                self.reddit.subreddit.create(new_name, title=None,
+                                             link_type='any', **kwargs)
+            assert excinfo.value.error_type == 'NO_TEXT'
+
+            with pytest.raises(APIException) as excinfo:
```

Comment:
One reason for the separate tests is that I look through the cassettes when I add them to ensure there aren't any unexpected requests. I started reviewing the cassette first and was surprised to see redirects in it for this test. Then I realized what was going on :).

Overall, I really like the different test cases you have, and they will be perfect in their own test function.